From 256678d445bc9aa86bbbc38f2a3537bc725c4a55 Mon Sep 17 00:00:00 2001
From: Dragan Marinkovic <dmarinkovi@sierrawireless.com>
Date: Mon, 18 Nov 2019 15:22:55 -0800
Subject: [PATCH] Do not use real lckpwdf()

shadow utility is using lckpwdf() method from libc. This
represents the problem, because lckpwdf() is storing
/etc/shadow related lock file in /etc, and if /etc is not
writable lock could not be obtain and password change
operation will fail.

The code which does not depend on lckpwdf(), stores
lock file at the location of shadow file. This approach
is much more suitable in cases /etc/ is not writable.

Resolves: LXSWIREF-1617
Related: LXSWIREF-1579
Signed-off-by: Dragan Marinkovic <dmarinkovi@sierrawireless.com>
---
 commonio.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/commonio.c b/lib/commonio.c
index 82cccb9..50f31c4 100644
--- a/lib/commonio.c
+++ b/lib/commonio.c
@@ -399,7 +399,7 @@ int commonio_lock_nowait (struct commonio_db *db, bool log)
 
 int commonio_lock (struct commonio_db *db)
 {
-#ifdef HAVE_LCKPWDF
+#if defined(HAVE_LCKPWDF) && !defined(DISABLE_REAL_LCKPWDF)
 	/*
 	 * only if the system libc has a real lckpwdf() - the one from
 	 * lockpw.c calls us and would cause infinite recursion!
@@ -470,7 +470,7 @@ static void dec_lock_count (void)
 				nscd_flush_cache ("group");
 				nscd_need_reload = false;
 			}
-#ifdef HAVE_LCKPWDF
+#if defined(HAVE_LCKPWDF) && !defined(DISABLE_REAL_LCKPWDF)
 			ulckpwdf ();
 #endif				/* HAVE_LCKPWDF */
 		}
-- 
2.20.1

